### PR TITLE
Multiclass identification in Dixon and minor cleanup (by Alexander Hulpke)

### PR DESCRIPTION
**Pull request by fingolfin**
This contains those changes from PR #5394 which I think are uncontroversial, and so once the CI for this PR passes, I am happy to merge right away.

### CODE REVIEW

File: lib/ringsc.gi [hunk 1]
```diff
@@ -917,11 +917,25 @@ end);
...
 function(R,e)
-  if not IsFinite(R) then
+local o,pow,a;
+  if not IsFinite(R) or One(R)=fail then
     TryNextMethod();
   fi;
-  return One(R)<>fail
-         and ForAny(Enumerator(R),x->x*e=One(R) and e*x=One(R));
+
+  # compute powers of e until we reach one or repeat
```
Comment (ThomasBreuer) [inline lib/ringsc.gi:925]: The previous code checked whether there is an element that is both a left and a right inverse.
The proposed code checks for a power that is the identity element, which is sufficient if the multiplication is associative, but why is it correct in nonassociative situations? The code establishes a left inverse, but why is it a right inverse?

File: lib/ghomperm.gi [hunk 2]
```diff
@@ -1074,6 +1074,8 @@ InstallMethod( CompositionMapping2, "group hom. with perm group hom.",
     function( hom1, hom2 )
     local   prd,  stb,  levs,  S,t,i,oli;
 
+    Size(Source(hom2));
+    Size(Range(hom2));
     stb := StructuralCopy( StabChainMutable( hom2 ) );
```
Comment (ThomasBreuer) [inline lib/ghomperm.gi:1079]: A comment why the `Size` computations are useful would be helpful.
I would have guessed that `StabChainMutable` will compute the orders of source and range as soon as the values are interesting.

File: lib/ctblperm.gi [hunk 3]
```diff
@@ -213,16 +220,22 @@ local k,structures,ambiguousStructures,i,j,p,cem,ces,z,t,cen,a,
...
   D.ids:=[];
+  D.chids:=[];
   D.rids:=[];
+  D.canreps:=[];
   for i in [1..D.klanz] do
     D.ids[i]:=D.identification(D,D.classreps[i]);
+    D.chids[i]:=D.cheapIdentification(D,D.classreps[i]);
     D.rids[i]:=
-     D.rationalidentification(D,D.classreps[i]);
+    D.rationalidentification(D,D.classreps[i]);
```
Comment (ThomasBreuer) [inline lib/ctblperm.gi:230]: Shouldn't the line be indented by one more space than before?

**Review by ThomasBreuer — COMMENTED** [review]
I have added some comments.

File: lib/ringsc.gi (hunk 1 above)
Comment (hulpke) [reply lib/ringsc.gi:925]: Ah. My world is associative. Will fix.

**Review by hulpke — COMMENTED** [review]
With Thomas remark this requires changes. I will do so in my *original* PR (because that is the one I can change)

File: lib/ctblperm.gi (hunk 3 above)
Comment (hulpke) [reply lib/ctblperm.gi:230]: Will indent (in original PR)

File: lib/ringsc.gi (hunk 1 above)
Comment (hulpke) [reply lib/ringsc.gi:925]: Fixed in original

**Comment by fingolfin** [comment]
I have backported the fixes here plus two more commits from #5394

**Review by hulpke — APPROVED** [review]

**Pull request merged**